gluster-exporter/metric_volume.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"github.com/gluster/gluster-prometheus/pkg/glusterutils"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	volumeLabels = []string{
+		"volume",
+	}
+
+	glusterVolumeTotalCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "gluster",
+			Name:      "volume_total_count",
+			Help:      "Total no of volumes",
+		},
+		[]string{},
+	)
+	glusterVolumeCreatedCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "gluster",
+			Name:      "volume_created_count",
+			Help:      "Freshly created no of volumes",
+		},
+		[]string{},
+	)
+	glusterVolumeStartedCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "gluster",
+			Name:      "volume_started_count",
+			Help:      "Total no of started volumes",
+		},
+		[]string{},
+	)
+	glusterVolumeBrickCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "gluster",
+			Name:      "volume_brick_count",
+			Help:      "Total no of bricks in volume",
+		},
+		volumeLabels,
+	)
+)
+
+func getVolumeLabels(volname string) prometheus.Labels {
+	return prometheus.Labels{
+		"volume": volname,
+	}
+}
+
+func VolumeCounts() {
+	volumes, _ := glusterutils.VolumeInfo(&glusterConfig)
+
+	volCount, volStartCount, volCreatedCount := 0, 0, 0
+
+	volCount = len(volumes)
+	for _, volume := range volumes {
+		switch state := volume.State; state {
+		case "Started":
+			volStartCount += 1
+		case "Created":
+			volCreatedCount += 1
+		default:
+			// Volume is just created, nothing to do
+		}
+		volBrickCount := 0
+		for _, subvol := range volume.SubVolumes {
+			volBrickCount += len(subvol.Bricks)
+		}
+		glusterVolumeBrickCount.With(getVolumeLabels(volume.Name)).Set(float64(volBrickCount))
+	}
+	glusterVolumeTotalCount.With(prometheus.Labels{}).Set(float64(volCount))
+	glusterVolumeStartedCount.With(prometheus.Labels{}).Set(float64(volStartCount))
+	glusterVolumeCreatedCount.With(prometheus.Labels{}).Set(float64(volCreatedCount))
+}
+
+func init() {
+	prometheus.MustRegister(glusterVolumeTotalCount)
+	prometheus.MustRegister(glusterVolumeStartedCount)
+	prometheus.MustRegister(glusterVolumeCreatedCount)
+	prometheus.MustRegister(glusterVolumeBrickCount)
+
+
+	// Register to update this every 2 seconds
```

Comment:
nit: comment is out of sync

Reply:
Will change this or remove if not required